File: 2026/b.py
#!/usr/bin/env python3

# ICPC Asia Japan 2026 Online First-Round Contest
# Problem B: Vending machines

def solve(xs, d):
    n = len(xs)
    nv = 0
    v = -d - 1  # 番兵: 最後の自動販売機
    for i in range(n):
        # 最後の自動販売機でカバーできない
        if v + d < xs[i]:
            nv += 1
            v = min(xs[i] + d, 10 ** 8)
    return nv

while True:
    n, d = map(int, input().split())
    if n == 0:
        break
    xs = list(map(int, input().split()))
    assert len(xs) == n
    print(solve(xs, d))

File: 2026/b.scala
#!/usr/bin/env scala

// ICPC Asia Japan 2026 Online First-Round Contest
// Problem B: Vending machines

import scala.io.StdIn

def solve(xs: Array[Int], d: Int): Int = {
  val v = -(d + 1)  // sentinel: the last vending machine
  xs.foldLeft((v, 0))((acc, x) => {
    val (v, nv) = acc  // pos-of-vending-machine, #-of-vending-machine
    if (v + d < x) (x + d, nv + 1) else (v, nv)
  })._2
}

def main(): Unit = {
    val Array(n, d) = inputIntArray()
    if (n != 0) {
      val xs = inputIntArray()
      assert(xs.length == n)

      println(solve(xs, d))
      main()
    }
}

def inputIntArray(): Array[Int] =
  StdIn.readLine().split(' ').map(_.toInt)

main()
